src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/CaseDataAccessControl.java

```diff
@@ -20,5 +23,5 @@ default void grantAccess(String caseId, String idamUserId) {
 
     }
 
-    CaseAccessMetadata generateAccessMetadata(String caseId);
+    CaseAccessMetadata generateAccessMetadata(String caseId, boolean isCheckingRegionLocationFilteringChecks);
```

Comment:
Can we expose multiple methods instead of boolean flag in the interface. 
If it's deeply nested we can at least hide under the implementation class.

Reply:
So a `generateAccessMetaDataForCreate` method or something similar ?

Reply:
Fixed in https://github.com/hmcts/ccd-data-store-api/pull/1544/commits/19487d5e45f1ee364455c0509f8ee8c3348de2cb